homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -52,33 +57,45 @@
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the Enphase Envoy sensor."""
-
     ip_address = config[CONF_IP_ADDRESS]
     monitored_conditions = config[CONF_MONITORED_CONDITIONS]
     name = config[CONF_NAME]
+    username = config[CONF_USERNAME]
+    password = config[CONF_PASSWORD]
+
+    envoy_reader = EnvoyReader(ip_address, username, password)
 
     entities = []
     # Iterate through the list of sensors
     for condition in monitored_conditions:
         if condition == "inverters":
-            inverters = await EnvoyReader(ip_address).inverters_production()
-            if isinstance(inverters, dict):
-                for inverter in inverters:
-                    entities.append(
-                        Envoy(
-                            ip_address,
-                            condition,
-                            f"{name}{SENSORS[condition][0]} {inverter}",
-                            SENSORS[condition][1],
+            try:
+                inverters = await envoy_reader.inverters_production()
+                if isinstance(inverters, dict):
+                    for inverter in inverters:
+                        entities.append(
+                            Envoy(
+                                envoy_reader,
+                                condition,
+                                f"{name}{SENSORS[condition][0]} {inverter}",
+                                SENSORS[condition][1],
+                                True,
+                            )
                         )
-                    )
+            except requests.exceptions.HTTPError:
```

Comment:
Please only wrap the line that can raise and `continue` if that happens.

We'll also save indentation that way.

Reply:
I tried to keep the `_LOGGER.warning` line in a single line, but when I did the commit the line was wrapped. :man_shrugging:

Originally I had this:
```
except requests.exceptions.HTTPError:
    _LOGGER.warning("Authentication for Inverter data failed during setup: %s", ip_address)
```

Then during the commit, the line was changed to:
```
except requests.exceptions.HTTPError:
    _LOGGER.warning(
        "Authentication for Inverter data failed during setup: %s",
        ip_address,
    )
```

I'll add a `continue`

Reply:
I mean that we should move the except up under the api call to only wrap a single line.

Reply:
Ok I understand.  I'll make that change.